lib/argonaut/bot.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Bot do
+  alias __MODULE__
+
+  @matchers [
+    {:reservation_action, ~r/(?<env>.*):(?<app>.*) (?<action>.*)/}
+  ]
+
+  @default_handler {:default_handler, nil}
+
+  def reply(message) when is_binary(message) do
+    {action, regex} = @matchers |> Enum.find(@default_handler, fn {_, re} ->
+      re |> Regex.match?(message)
+    end)
+
+    if action == :default_handler do
+      apply(__MODULE__, action, [])
+    else
+      matches = Regex.named_captures(regex, message)
+      apply(__MODULE__, action, [matches])
+    end
+  end
+
+  def reservation_action(%{"action" => action, "app" => app, "env" => env}) do
+    cond do
+      action == "reserve" -> "reservation"
+      action == "renew" -> "renew"
+      action == "release" -> "release"
+      action == "info" -> "info"
+    end
+  end
```

Comment:
```suggestion
def reservation_action(%{"action" => "reserve"}), do: "reservation"
def reservation_action(%{"action" => "renew"}), do: "renew"
def reservation_action(%{"action" => "release"}), do: "release"
def reservation_action(%{"action" => "info"}), do: "info"
    end
```

Doesn't look like you're currently using `app` or `env`

Reply:
He's not wrong :).